examples/hmcecs.py

```diff
@@ -0,0 +1,147 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Example: Hamiltonian Monte Carlo with Energy Conserving Subsampling
+=========================
+
+This example illustrates the use of subsampling of HMC using Energy Conserving Subsampling when the likelihood is
```

Comment:
maybe something like this?

This example illustrates the use of data subsampling in HMC using Energy Conserving Subsampling, which is applicable when the likelihood factorizes as a product of N terms.